### PR TITLE
Debug markers 

### PR DESCRIPTION
**Pull request by Mofef**
We added a couple of marker topics to visualize whats going on with submaps and constraints in cartographer. This also includes the constraint visualization I posted to the mailinglist some days/weeks ago.
On Monday Dominik will be back so he probably has time to work on your feedback.

### CODE REVIEW

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 1]
```diff
@@ -14,8 +14,8 @@
  * limitations under the License.
  */
 
+#include "cartographer/mapping/sparse_pose_graph.h"
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Style: Related header file comes first and separate from all others, i.e. this belongs one line down, separate by an empty line.

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 2]
```diff
@@ -105,13 +105,115 @@ cartographer_ros_msgs::SubmapList MapBuilderBridge::GetSubmapList() {
...
+
+    // creating residual error line strip marker (blue)
+    residual_error_marker.id = i++;
+    residual_error_marker.type = visualization_msgs::Marker::LINE_STRIP;
+    residual_error_marker.header.stamp = ::ros::Time::now();
+    residual_error_marker.header.frame_id = options_.map_frame;
+    residual_error_marker.color.a = 1.0;
+    residual_error_marker.scale.x = 0.02;
+
+    // creating points for the line strip markers
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Style: Use full sentences in comments, i.e. capitalize at the beginning and end with a period.

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 3]
```diff
@@ -105,13 +105,115 @@ cartographer_ros_msgs::SubmapList MapBuilderBridge::GetSubmapList() {
...
+    geometry_msgs::Point submap_point, submap_pose_point, trajectory_node_point;
+
+    submap_point.x = submap_transforms[constraint.i].translation().x();
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Add a function `ToGeometryMsgPoint()` to `msg_conversion.cc`?

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 4]
```diff
@@ -105,13 +105,115 @@ cartographer_ros_msgs::SubmapList MapBuilderBridge::GetSubmapList() {
...
+  const std::vector<cartographer::transform::Rigid3d> submap_transforms =
+      map_builder_.sparse_pose_graph()->GetSubmapTransforms();
+
+  const auto trajectory_nodes = map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
+  const auto constraints = map_builder_.sparse_pose_graph()->constraints();
+
+  int i = 0;
+  for (const auto &constraint : constraints) {
+
+	visualization_msgs::Marker constraint_marker, residual_error_marker;
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Please run `clang-format --style=Google`.

**Comment by Mofef** [comment]
@jihoonl will open a new cleaned up version of this PR. So I close

**Pull request closed without merging**